src/uu/od/src/parse_formats.rs

```diff
@@ -290,7 +294,50 @@ fn parse_type_string(params: &str) -> Result<Vec<ParsedFormatterItemInfo>, Strin
 
         let mut byte_size = 0u8;
         let mut show_ascii_dump = false;
-        if is_format_size_char(ch, type_cat, &mut byte_size) {
+        let mut float_variant = None;
+        if type_cat == FormatTypeCategory::Float {
+            match ch {
+                Some('B') => {
+                    byte_size = 2;
+                    float_variant = Some('B');
+                    ch = chars.next();
+                }
+                Some('H') => {
+                    byte_size = 2;
+                    float_variant = Some('H');
+                    ch = chars.next();
+                }
```

Comment:
We can factorize here

```suggestion
                Some(var @ 'B' | 'H') => {
                    byte_size = 2;
                    float_variant = Some(var);
                    ch = chars.next();
                }
```

Reply:
Thanks for the suggestion! I’ve consolidated the two Some('B') and Some('H') into a single.